Add new section for test integrations

LeoColman opened the pull request:
This commit adds documentation on how Testcontainers can be integrated with the Kotest framework.


From the very old #2533 it was decided that a reference to Kotest would be great in the docs. This commit adds it.

eddumelendez commented on docs/test_framework_integration/kotest.md:
```diff
@@ -0,0 +1,62 @@
+
+[Kotest](https://github.com/kotest/kotest) extension for [Testcontainers](https://github.com/testcontainers/testcontainers-java) library, which allows to use Docker containers inside of Kotest tests.
+
+The source code is maintained at [Kotest Extensions TestContainers](https://github.com/kotest/kotest-extensions-testcontainers)
```
```suggestion
The source code is maintained at [Kotest Extensions Testcontainers](https://github.com/kotest/kotest-extensions-testcontainers)
```

eddumelendez commented on docs/test_framework_integration/kotest.md:
```diff
@@ -0,0 +1,62 @@
+
+## Documentation
+
+The full and updated documentation for Kotest integration with TestContainers can be found at [kotest.io](https://kotest.io/docs/extensions/test_containers.html)
```
```suggestion
The full and updated documentation for Kotest integration with Testcontainers can be found at [kotest.io](https://kotest.io/docs/extensions/test_containers.html)
```

eddumelendez reviewed (changes requested):
Thanks for the contribution! I've left some comments

LeoColman commented:
Done :)

eddumelendez commented:
Hi @LeoColman, thinking more about this... I think we should not duplicate the existing documentation in kotest but we should point to it. what do you think about renaming the file to integration and then add Kotest with a link to https://kotest.io/docs/extensions/test_containers.html

The content can says something like `The following open source projects use Testcontainers`

LeoColman commented:
I agree!

I even tried to make it smaller to avoid maintenance issues.

How could I apply your suggestion?

Apr 20, 2023 19:46:14 Eddú Meléndez Gonzales ***@***.***>:

> 
> Hi @LeoColman[https://github.com/LeoColman], thinking more about this... I think we should not duplicate the existing documentation in kotest but we should point to it. what do you think about renaming the file to integration and then add Kotest with a link to https://kotest.io/docs/extensions/test_containers.html
> 
> The content can says something like *The following open source projects use Testcontainers*
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/testcontainers/testcontainers-java/pull/6930#issuecomment-1517066159], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAMBCI7FIZCNKMDDT3MKGLLXCHDEDANCNFSM6AAAAAAXEV2VXY].
> You are receiving this because you were mentioned.[Tracking image][https://github.com/notifications/beacon/AAMBCI6NH45AMXKBGEDNKN3XCHDEDA5CNFSM6AAAAAAXEV2VX2WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTS2NSL26.gif]
>

eddumelendez commented:
Just update this PR :)

LeoColman commented:
Alright, I renamed it to `external` and replaced the documentation with links to the module itself.


Also took the opportunity to add another integration from #3298 to the same page. <-- @jlink @gtrefs

LeoColman commented:
I thought of adding a `last updated` or `last maintained` to each of them, to prevent accidentally using a very old framework. Let me know what you think @eddumelendez

eddumelendez commented on docs/test_framework_integration/external.md:
```diff
@@ -0,0 +1,10 @@
+
+The following Open Source frameworks add direct integration to Testcontainers
+
+
```
```suggestion
```

eddumelendez commented:
Thanks for your contribution, @LeoColman ! This will be available soon!
